gnovm/cmd/gno/repl.go

```diff
@@ -88,91 +89,71 @@ func execRepl(cfg *replCfg, args []string) error {
 //   gno> import "gno.land/p/demo/avl"     // import the p/demo/avl package
 //   gno> func a() string { return "a" }   // declare a new function named a
 //   gno> /src                             // print current generated source
-//   gno> /editor                          // enter in editor mode to add several lines
 //   gno> /reset                           // remove all previously inserted code
 //   gno> println(a())                     // print the result of calling a()
-//   gno> /exit
+//   gno> /exit                            // alternative to <Ctrl-D>
 `)
 	}
 
 	return runRepl(cfg)
 }
 
 func runRepl(cfg *replCfg) error {
-	// init repl state
 	r := repl.NewRepl()
 
 	if cfg.initialCommand != "" {
 		handleInput(r, cfg.initialCommand)
 	}
 
-	var multiline bool
-	for {
-		fmt.Fprint(os.Stdout, "gno> ")
+	fmt.Fprint(os.Stdout, "gno> ")
 
-		input, err := getInput(multiline)
-		if err != nil {
-			return err
+	prevline := ""
+	liner := bufio.NewScanner(os.Stdin)
+
+	for liner.Scan() {
+		line := liner.Text()
+		if prevline != "" {
+			line = prevline + "\n" + line
+			prevline = ""
+		}
+
+		if err := handleInput(r, line); err != nil {
+			var goScanError scanner.ErrorList
+			if errors.As(err, &goScanError) {
+				// We assume that a Go scanner error indicates an incomplete Go statement.
+				// Append next line and retry.
+				prevline = line
+			} else {
+				fmt.Fprintln(os.Stderr, err)
+			}
 		}
 
-		multiline = handleInput(r, input)
+		if prevline == "" {
+			fmt.Fprint(os.Stdout, "gno> ")
+		} else {
+			fmt.Fprint(os.Stdout, "... ")
```

Comment:
I'm visually very disturbed that the first line of the repl and the following are not aligned.

Any chance we can change this to `"...  "`? (5 characters, like `"gno> "`)

Reply:
done